src/aooptionsdialog.cpp

```diff
@@ -1,596 +1,743 @@
-    ui_audio_layout->setLabelAlignment(Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter);
-    ui_audio_layout->setFormAlignment(Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop);
-    ui_audio_layout->setContentsMargins(0, 0, 0, 0);
+  ui_gameplay_form->setWidget(8, QFormLayout::LabelRole, ui_ms_lbl);
+
+  QSettings *configini = ao_app->configini;
+  ui_ms_textbox = new QLineEdit(ui_form_layout_widget);
+  ui_ms_textbox->setText(configini->value("master", "").value<QString>());
+
+  ui_gameplay_form->setWidget(8, QFormLayout::FieldRole, ui_ms_textbox);
+
+  ui_discord_lbl = new QLabel(ui_form_layout_widget);
+  ui_discord_lbl->setText(tr("Discord:"));
+  ui_discord_lbl->setToolTip(
+      tr("Allows others on Discord to see what server you are in, "
+         "what character are you playing, and how long you have "
+         "been playing for."));
+
+  ui_gameplay_form->setWidget(9, QFormLayout::LabelRole, ui_discord_lbl);
+
+  ui_discord_cb = new QCheckBox(ui_form_layout_widget);
+  ui_discord_cb->setChecked(ao_app->is_discord_enabled());
+
+  ui_gameplay_form->setWidget(9, QFormLayout::FieldRole, ui_discord_cb);
+
+  ui_epilepsy_lbl = new QLabel(ui_form_layout_widget);
+  ui_epilepsy_lbl->setText(tr("Allow Shake/Flash:"));
+  ui_epilepsy_lbl->setToolTip(
+      tr("Allows screenshaking and flashing. Disable this if you have concerns "
+         "or issues with photosensitivity and/or seizures."));
+
+  ui_gameplay_form->setWidget(10, QFormLayout::LabelRole, ui_epilepsy_lbl);
+
+  ui_epilepsy_cb = new QCheckBox(ui_form_layout_widget);
+  ui_epilepsy_cb->setChecked(ao_app->is_shakeandflash_enabled());
+
+  ui_gameplay_form->setWidget(10, QFormLayout::FieldRole, ui_epilepsy_cb);
+
+  ui_language_label = new QLabel(ui_form_layout_widget);
+  ui_language_label->setText(tr("Language:"));
+  ui_language_label->setToolTip(
+      tr("Sets the language if you don't want to use your system language."));
+  ui_gameplay_form->setWidget(11, QFormLayout::LabelRole, ui_language_label);
+
+  ui_epilepsy_lbl = new QLabel(ui_form_layout_widget);
+  ui_epilepsy_lbl->setText(tr("Allow Shake/Flash:"));
+  ui_epilepsy_lbl->setToolTip(
+      tr("Allows screenshaking and flashing. Disable this if you have concerns "
+         "or issues with photosensitivity and/or seizures."));
+
+  ui_gameplay_form->setWidget(11, QFormLayout::LabelRole, ui_epilepsy_lbl);
+
+  ui_epilepsy_cb = new QCheckBox(ui_form_layout_widget);
+  ui_epilepsy_cb->setChecked(ao_app->is_shakeandflash_enabled());
+
+  ui_gameplay_form->setWidget(11, QFormLayout::FieldRole, ui_epilepsy_cb);
+
+  ui_language_combobox = new QComboBox(ui_form_layout_widget);
+  ui_language_combobox->addItem(
+      configini->value("language", "  ").value<QString>() +
+      " - Keep current setting");
+  ui_language_combobox->addItem("   - Default");
+  ui_language_combobox->addItem("en - English");
+  ui_language_combobox->addItem("de - Deutsch");
+  ui_language_combobox->addItem("es - Español");
```

Comment:
Spanish option added twice and portuguese missing, see https://github.com/AttorneyOnline/AO2-Client/blob/901012eb2872ad5d4af23b82cc6dc0d1da587b20/src/aooptionsdialog.cpp#L191